### PR TITLE
Promisified Guild#fetchAllMembers

### PR DESCRIPTION
**Pull request by LJNeon**
I did a basic test to make sure this works, more testing may or may not be useful. I used the `requestTimeout` client option for this, as I don't believe that making a new option is worth it here and it seemed to be the best fitting of all existing options. I also never reject the promise as the point here was to mirror previous behavior, only with a Promise that resolves at the end. I'm open to feedback on those two points.

### CODE REVIEW

File: lib/gateway/Shard.js [hunk 1]
```diff
@@ -144,6 +144,15 @@ class Shard extends EventEmitter {
         this.connecting = false;
         this.ready = false;
         this.preReady = false;
+        if(this.requestMembersPromise !== undefined) {
+            for(const guildID in this.requestMembersPromise) {
+                if(!this.requestMembersPromise.hasOwnProperty(guildID))
```
Comment (apacheli) [inline lib/gateway/Shard.js]: Missing curly

**Review by apacheli — CHANGES_REQUESTED** [review]

File: lib/gateway/Shard.js (hunk 1 above)
Comment (LJNeon) [reply lib/gateway/Shard.js]: Fixed

File: lib/gateway/Shard.js [hunk 2]
```diff
@@ -144,6 +144,16 @@ class Shard extends EventEmitter {
         this.connecting = false;
         this.ready = false;
         this.preReady = false;
+        if(this.requestMembersPromise !== undefined) {
+            for(const guildID in this.requestMembersPromise) {
+                if(!this.requestMembersPromise.hasOwnProperty(guildID)) {
+                    continue;
+                }
+                clearTimeout(this.requestMembersPromise[guildID].timeout);
+                this.requestMembersPromise[guildID].res();
```
Comment (abalabahaha) [inline lib/gateway/Shard.js]: `.received` should be returned here too

File: lib/structures/Guild.js [hunk 3]
```diff
@@ -149,9 +149,11 @@ class Guild extends Base {
 
     /**
     * Request all guild members from Discord
+    * @arg {Number} [timeout] The number of milliseconds to wait before resolving early. Defaults to the `requestTimeout` client option
+    * @returns {Promise<Number>} Resolves with the total amount of fetched members.
```
Comment (abalabahaha) [inline lib/structures/Guild.js]: Nit: `total number of`

**Review by abalabahaha — CHANGES_REQUESTED** [review]

File: lib/structures/Guild.js (hunk 3 above)
Comment (LJNeon) [reply lib/structures/Guild.js]: should be fixed

File: lib/gateway/Shard.js (hunk 2 above)
Comment (LJNeon) [reply lib/gateway/Shard.js]: should be fixed

**Review by abalabahaha — APPROVED** [review]

**Pull request merged**